### DISCUSSION TITLE
Adding SoundScapeExplorer to the Bioacoustics Software Database

### DISCUSSION THREAD

**Discussion by rhine3** [sound-scape-explorer/sound-scape-explorer#239] [General]
Hi! A colleague recently shared SoundScapeExplorer and CoralSoundExplorer with me. I would like to add them to a public database I maintain of bioacoustics software ([GitHub](https://github.com/rhine3/bioacoustics-software), [website](https://rhine3.github.io/bioacoustics-software/)).

I've looked through the docs website of SSE, and it looks like it's a well-made software, but I'm still not sure I understand its functionality, such as:
- what are user labels and intervals? 
- what metrics can be defined?
- what kind of data can you extract with neural networks? can you train your own neural networks?
- what kind of "spaces" can be dimensionally reduced; spectrograms?
- what is the difference or relationship between SSE and CSE?

I would love to include SSE, CSE, or both in the database, so if you could please help me answer these questions, I would appreciate it! 

Thanks!

**Comment by bamdadfr** [comment]
Hi @rhine3,

Thanks for the interest into our work! I will try to speak on behalf of the team but will eventually need to ask the product owners for the definitive approval.

In the meantime, I will be glad to give you insights about the questions you asked:

> what are user labels and intervals?

Users can attach additional text information to their audio files described in the configuration file (.xlsx for now). These are called *user labels* for lack of a better name.

During the audio file processing, data gets extracted usually at a 1s rate. Then it gets *aggregated* according to the user settings. If an *integration* of 15 seconds is requested, then 15 blocks of 1s data is merged together (`np.mean()` for now as each extractor might implement its own *aggregation* logic in the future). The *intervals* are all of these output blocks of 15s.

> what metrics can be defined?

Documentation is a bit light here. The metrics are the following:
![image](https://github.com/user-attachments/assets/57dcd499-9062-4ebb-8317-281a3237169d)
![image](https://github.com/user-attachments/assets/be592e2c-42c3-4e03-8ca8-01686a8a14d5)

> what kind of data can you extract with neural networks? can you train your own neural networks?

At the moment, both SSE and CSE implements only one NN extractor `VGGish`.
For each second of audio, an array of 128 numbers is generated by the network.

New neural network implementations are planned e.g. `BirdNET`.

> what kind of "spaces" can be dimensionally reduced; spectrograms?

Typically, users will reduce the `VGGish` output from 128 dimensions to 2 or 3 dimensions in order to visualise them.

> what is the difference or relationship between SSE and CSE?

From a technical standpoint, SSE is the upstream version. CSE is a *snapshot* of SSE version 12.3.0 as [described here](https://sound-scape-explorer.github.io/blog/2024/01/04/version-CSE) + minor fixes and improvements.

CSE is tied to a very specific project in Bora Bora. When the publishing process will be over, the CSE repository will most probably switch to read-only mode. We will then advise users to using SSE to get latest features. Also, you will probably need to cite the software accordingly (CSE).

---

Hope this all makes sense.

Feel free to let us know if you need further information.

Regards,
Bamdad

**Reply by rhine3** [reply]
Thanks so much, Bamdad, your explanations cleared up the questions I had! I have added SoundScapeExplorer to the database and it should be visible online shortly.
